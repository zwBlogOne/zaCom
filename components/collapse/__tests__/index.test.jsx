import React from 'react';
import { render, mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Collapse from '../index';

describe('Collapse', () => {
  const props = {};

  it('renders correctly', () => {
    const wrapper = render(
      <Collapse {...props}>
        <Collapse.Item title="50元套餐">
          <div>50元套餐内容</div>
        </Collapse.Item>
      </Collapse>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders correctly with collapse mode', () => {
    props.multiple = false;
    const wrapper = render(
      <Collapse {...props}>
        <Collapse.Item title="50元套餐">
          <div>50元套餐内容</div>
        </Collapse.Item>
      </Collapse>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders correctly with animated', () => {
    props.animated = true;
    const wrapper = render(
      <Collapse {...props}>
        <Collapse.Item title="50元套餐">
          <div>50元套餐内容</div>
        </Collapse.Item>
      </Collapse>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders correctly with defaultActiveIndex', () => {
    props.defaultActiveIndex = [1];
    const wrapper = render(
      <Collapse {...props}>
        <Collapse.Item title="50元套餐">
          <div>50元套餐内容</div>
        </Collapse.Item>
        <Collapse.Item title="100元套餐">
          <div>100元套餐内容</div>
        </Collapse.Item>
      </Collapse>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders correctly with string defaultActiveIndex', () => {
    props.defaultActiveIndex = '1';
    const wrapper = render(
      <Collapse {...props}>
        <Collapse.Item title="50元套餐">
          <div>50元套餐内容</div>
        </Collapse.Item>
        <Collapse.Item title="100元套餐">
          <div>100元套餐内容</div>
        </Collapse.Item>
      </Collapse>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders correctly with number defaultActiveIndex', () => {
    props.defaultActiveIndex = 0;
    const wrapper = render(
      <Collapse {...props}>
        <Collapse.Item title="50元套餐">
          <div>50元套餐内容</div>
        </Collapse.Item>
        <Collapse.Item title="100元套餐">
          <div>100元套餐内容</div>
        </Collapse.Item>
      </Collapse>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders correctly with activeIndex', () => {
    props.activeIndex = [0];
    const wrapper = render(
      <Collapse {...props}>
        <Collapse.Item title="50元套餐">
          <div>50元套餐内容</div>
        </Collapse.Item>
        <Collapse.Item title="100元套餐">
          <div>100元套餐内容</div>
        </Collapse.Item>
      </Collapse>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders correctly with string activeIndex', () => {
    props.activeIndex = '1';
    const wrapper = render(
      <Collapse {...props}>
        <Collapse.Item title="50元套餐">
          <div>50元套餐内容</div>
        </Collapse.Item>
        <Collapse.Item title="100元套餐">
          <div>100元套餐内容</div>
        </Collapse.Item>
      </Collapse>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders correctly with number activeIndex', () => {
    props.activeIndex = 0;
    const wrapper = render(
      <Collapse {...props}>
        <Collapse.Item title="50元套餐">
          <div>50元套餐内容</div>
        </Collapse.Item>
        <Collapse.Item title="100元套餐">
          <div>100元套餐内容</div>
        </Collapse.Item>
      </Collapse>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders correctly with dynamic activeIndex', () => {
    props.activeIndex = [0];
    const wrapper = shallow(
      <Collapse {...props}>
        <Collapse.Item title="50元套餐">
          <div>50元套餐内容</div>
        </Collapse.Item>
        <Collapse.Item title="100元套餐">
          <div>100元套餐内容</div>
        </Collapse.Item>
      </Collapse>
    );
    wrapper.setProps({ activeIndex: '1' });

    expect(wrapper.state('activeIndex')).toEqual(['1']);
  });

  it('click collapse item correctly', () => {
    props.onChange = jest.fn();
    const wrapper = mount(
      <Collapse {...props}>
        <Collapse.Item title="50元套餐" {...props}>
          <div>50元套餐内容</div>
        </Collapse.Item>
      </Collapse>
    );
    wrapper.find('.za-collapse-item-title').simulate('click');
    expect(props.onChange).toBeCalled();
  });

  it('renders correctly with open mode', () => {
    props.open = true;
    const wrapper = render(
      <Collapse {...props}>
        <Collapse.Item title="50元套餐">
          <div>50元套餐内容</div>
        </Collapse.Item>
        <Collapse.Item title="100元套餐">
          <div>100元套餐内容</div>
        </Collapse.Item>
      </Collapse>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('click collapse item correctly with open mode', () => {
    props.onChange = jest.fn();
    const wrapper = mount(
      <Collapse {...props}>
        <Collapse.Item title="50元套餐">
          <div>50元套餐内容</div>
        </Collapse.Item>
      </Collapse>
    );
    wrapper.find('.za-collapse-item-title').simulate('click');
    expect(props.onChange).not.toBeCalled();
  });
});
