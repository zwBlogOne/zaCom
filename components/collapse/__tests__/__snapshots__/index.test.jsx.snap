// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collapse renders correctly 1`] = `
<div
  class="za-collapse"
>
  <div
    class="za-collapse-item"
  >
    <div
      class="za-collapse-item-title"
    >
      <div>
        50元套餐
      </div>
      <div
        class="za-collapse-item-arrow"
      />
    </div>
    <div
      class="za-collapse-item-content"
    >
      <div
        class="za-collapse-item-content-inner"
      >
        <div>
          50元套餐内容
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Collapse renders correctly with activeIndex 1`] = `
<div
  class="za-collapse"
>
  <div
    class="za-collapse-item active"
  >
    <div
      class="za-collapse-item-title"
    >
      <div>
        50元套餐
      </div>
      <div
        class="za-collapse-item-arrow"
      />
    </div>
    <div
      class="za-collapse-item-content za-collapse-item-content-anim"
    >
      <div
        class="za-collapse-item-content-inner"
      >
        <div>
          50元套餐内容
        </div>
      </div>
    </div>
  </div>
  <div
    class="za-collapse-item"
  >
    <div
      class="za-collapse-item-title"
    >
      <div>
        100元套餐
      </div>
      <div
        class="za-collapse-item-arrow"
      />
    </div>
    <div
      class="za-collapse-item-content za-collapse-item-content-anim"
    >
      <div
        class="za-collapse-item-content-inner"
      >
        <div>
          100元套餐内容
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Collapse renders correctly with animated 1`] = `
<div
  class="za-collapse"
>
  <div
    class="za-collapse-item"
  >
    <div
      class="za-collapse-item-title"
    >
      <div>
        50元套餐
      </div>
      <div
        class="za-collapse-item-arrow"
      />
    </div>
    <div
      class="za-collapse-item-content za-collapse-item-content-anim"
    >
      <div
        class="za-collapse-item-content-inner"
      >
        <div>
          50元套餐内容
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Collapse renders correctly with collapse mode 1`] = `
<div
  class="za-collapse"
>
  <div
    class="za-collapse-item"
  >
    <div
      class="za-collapse-item-title"
    >
      <div>
        50元套餐
      </div>
      <div
        class="za-collapse-item-arrow"
      />
    </div>
    <div
      class="za-collapse-item-content"
    >
      <div
        class="za-collapse-item-content-inner"
      >
        <div>
          50元套餐内容
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Collapse renders correctly with defaultActiveIndex 1`] = `
<div
  class="za-collapse"
>
  <div
    class="za-collapse-item"
  >
    <div
      class="za-collapse-item-title"
    >
      <div>
        50元套餐
      </div>
      <div
        class="za-collapse-item-arrow"
      />
    </div>
    <div
      class="za-collapse-item-content za-collapse-item-content-anim"
    >
      <div
        class="za-collapse-item-content-inner"
      >
        <div>
          50元套餐内容
        </div>
      </div>
    </div>
  </div>
  <div
    class="za-collapse-item active"
  >
    <div
      class="za-collapse-item-title"
    >
      <div>
        100元套餐
      </div>
      <div
        class="za-collapse-item-arrow"
      />
    </div>
    <div
      class="za-collapse-item-content za-collapse-item-content-anim"
    >
      <div
        class="za-collapse-item-content-inner"
      >
        <div>
          100元套餐内容
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Collapse renders correctly with number activeIndex 1`] = `
<div
  class="za-collapse"
>
  <div
    class="za-collapse-item active"
  >
    <div
      class="za-collapse-item-title"
    >
      <div>
        50元套餐
      </div>
      <div
        class="za-collapse-item-arrow"
      />
    </div>
    <div
      class="za-collapse-item-content za-collapse-item-content-anim"
    >
      <div
        class="za-collapse-item-content-inner"
      >
        <div>
          50元套餐内容
        </div>
      </div>
    </div>
  </div>
  <div
    class="za-collapse-item"
  >
    <div
      class="za-collapse-item-title"
    >
      <div>
        100元套餐
      </div>
      <div
        class="za-collapse-item-arrow"
      />
    </div>
    <div
      class="za-collapse-item-content za-collapse-item-content-anim"
    >
      <div
        class="za-collapse-item-content-inner"
      >
        <div>
          100元套餐内容
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Collapse renders correctly with number defaultActiveIndex 1`] = `
<div
  class="za-collapse"
>
  <div
    class="za-collapse-item active"
  >
    <div
      class="za-collapse-item-title"
    >
      <div>
        50元套餐
      </div>
      <div
        class="za-collapse-item-arrow"
      />
    </div>
    <div
      class="za-collapse-item-content za-collapse-item-content-anim"
    >
      <div
        class="za-collapse-item-content-inner"
      >
        <div>
          50元套餐内容
        </div>
      </div>
    </div>
  </div>
  <div
    class="za-collapse-item"
  >
    <div
      class="za-collapse-item-title"
    >
      <div>
        100元套餐
      </div>
      <div
        class="za-collapse-item-arrow"
      />
    </div>
    <div
      class="za-collapse-item-content za-collapse-item-content-anim"
    >
      <div
        class="za-collapse-item-content-inner"
      >
        <div>
          100元套餐内容
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Collapse renders correctly with open mode 1`] = `
<div
  class="za-collapse"
>
  <div
    class="za-collapse-item active"
  >
    <div
      class="za-collapse-item-title"
    >
      <div>
        50元套餐
      </div>
      <div
        class="za-collapse-item-arrow za-collapse-item-arrow-hidden"
      />
    </div>
    <div
      class="za-collapse-item-content za-collapse-item-content-anim"
    >
      <div
        class="za-collapse-item-content-inner"
      >
        <div>
          50元套餐内容
        </div>
      </div>
    </div>
  </div>
  <div
    class="za-collapse-item active"
  >
    <div
      class="za-collapse-item-title"
    >
      <div>
        100元套餐
      </div>
      <div
        class="za-collapse-item-arrow za-collapse-item-arrow-hidden"
      />
    </div>
    <div
      class="za-collapse-item-content za-collapse-item-content-anim"
    >
      <div
        class="za-collapse-item-content-inner"
      >
        <div>
          100元套餐内容
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Collapse renders correctly with string activeIndex 1`] = `
<div
  class="za-collapse"
>
  <div
    class="za-collapse-item"
  >
    <div
      class="za-collapse-item-title"
    >
      <div>
        50元套餐
      </div>
      <div
        class="za-collapse-item-arrow"
      />
    </div>
    <div
      class="za-collapse-item-content za-collapse-item-content-anim"
    >
      <div
        class="za-collapse-item-content-inner"
      >
        <div>
          50元套餐内容
        </div>
      </div>
    </div>
  </div>
  <div
    class="za-collapse-item active"
  >
    <div
      class="za-collapse-item-title"
    >
      <div>
        100元套餐
      </div>
      <div
        class="za-collapse-item-arrow"
      />
    </div>
    <div
      class="za-collapse-item-content za-collapse-item-content-anim"
    >
      <div
        class="za-collapse-item-content-inner"
      >
        <div>
          100元套餐内容
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Collapse renders correctly with string defaultActiveIndex 1`] = `
<div
  class="za-collapse"
>
  <div
    class="za-collapse-item"
  >
    <div
      class="za-collapse-item-title"
    >
      <div>
        50元套餐
      </div>
      <div
        class="za-collapse-item-arrow"
      />
    </div>
    <div
      class="za-collapse-item-content za-collapse-item-content-anim"
    >
      <div
        class="za-collapse-item-content-inner"
      >
        <div>
          50元套餐内容
        </div>
      </div>
    </div>
  </div>
  <div
    class="za-collapse-item active"
  >
    <div
      class="za-collapse-item-title"
    >
      <div>
        100元套餐
      </div>
      <div
        class="za-collapse-item-arrow"
      />
    </div>
    <div
      class="za-collapse-item-content za-collapse-item-content-anim"
    >
      <div
        class="za-collapse-item-content-inner"
      >
        <div>
          100元套餐内容
        </div>
      </div>
    </div>
  </div>
</div>
`;
